### PR TITLE
Adding examples and fixing the DSC related lint issues

### PR DESCRIPTION
**Pull request by vrdmr**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by fearthecowboy** [comment]
Hey @vrdmr ,

I was about to start reviewing the PR, but it still has an awful lot of validation errors. 

See: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/337677030

Can you review those, and see what you can do to get it to pass?

**Comment by vrdmr** [comment]
Hi @fearthecowboy,

Thanks. I'll take a look.

I am trying to run the model-validator but I am hitting the following:
```
varads-mbp:resource-manager varadmeru$ autorest --model-validator readme.md 
AutoRest code generation utility [version: 2.0.4245; node: v9.4.0]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/private/var/folders/sp/dpmjjqvj7gs3ylxw_08wpxk40000gn/T/.autorest/@microsoft.azure_autorest-core@2.0.4245/node_modules/@microsoft.azure/autorest-core/dist' (2.0.4245)
   Loading local AutoRest extension 'oav' (/usr/local/Cellar/node/9.4.0/lib/node_modules/oav)
FATAL: TypeError: Cannot read property 'ReadFile' of undefined
FATAL: swagger-document/model-validator - FAILED
FATAL: Error: Plugin model-validator reported failure.
Process() cancelled due to exception : Plugin model-validator reported failure.
  Error: Plugin model-validator reported failure.
varads-mbp:resource-manager varadmeru$ 
```

Could you please help me with this.

Thanks,
Varad

**Comment by fearthecowboy** [comment]
Hey @vrdmr , 

I'm seeing the problem using `--model-validator` too. I'm looking into it.

In the mean time, you can use the model-validator outside of autorest on individual swagger files:

Example 
``` powershell
# ensure the module globally
> npm install -g oav

# example: run it against a swagger file directly.
> oav validate-example Microsoft.Automation/stable/2015-10-31/dscCompilationJob.json
```

I'm looking into why using it as a plugin isn't working right now.

**Comment by fearthecowboy** [comment]
FYI @vrdmr 

This was the correct syntax (even tho it fails)
``` powershell
  
autorest --model-validator readme.md 
```

Using `--oav` and `--validation` don't do anything; the validators are activated with `--model-validator` and `--azure-validator` (you'll know which are correct, they will show you `Loading autorest extension` messages)

**Comment by fearthecowboy** [comment]
Hey @vrdmr,

I've asked the OAV author to fix the autorest extension, I think we've got a few glitches he has to take care of.

in the mean time, fall back to using oav manually on each of the files.

**Comment by vrdmr** [comment]
Hi Garrett,

I have fixed most of the DSC related issues and am working with other teams to fix their resources specs. Could you please take a look at this PR now and merge it in when you get a chance?

Thanks,
Varad

@fearthecowboy

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json [hunk 1]
```diff
@@ -264,10 +264,10 @@
       "properties": {
         "keyName": {
           "type": "string",
-          "description": "Gets or sets the agent registration key name - Primary or Secondary.",
+          "description": "Gets or sets the agent registration key name - primary or secondary.",
           "enum": [
-            "Primary",
-            "Secondary"
+            "primary",
```
Comment (fearthecowboy) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json]: !

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json [hunk 2]
```diff
@@ -1331,11 +1332,7 @@
           "description": "Gets or sets the etag of the resource."
         }
       },
-      "allOf": [
-        {
-          "$ref": "#/definitions/Resource"
-        }
-      ],
+      "x-ms-azure-resource": true,
```
Comment (fearthecowboy) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json]: Why has this `allOf` been removed?

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json [hunk 3]
```diff
@@ -1492,11 +1489,7 @@
           }
         }
       },
-      "allOf": [
-        {
-          "$ref": "#/definitions/Resource"
-        }
-      ],
+      "x-ms-azure-resource": true,
```
Comment (fearthecowboy) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json]: And this one?

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/dscCompilationJob.json [hunk 4]
```diff
@@ -203,6 +213,63 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/compilationjobs/{jobId}/streams/": {
```
Comment (fearthecowboy) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/dscCompilationJob.json]: Is this adding a new API or documenting an existing one that wasn't listed before?

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/dscNodeConfiguration.json [hunk 5]
```diff
@@ -189,12 +194,6 @@
               "$ref": "./definitions.json#/definitions/DscNodeConfiguration"
             }
           },
-          "201": {
```
Comment (fearthecowboy) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/dscNodeConfiguration.json]: Why was this deleted? This may result in a breaking change in the generated clients.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/dscCompilationJob.json [hunk 6]
```diff
@@ -107,6 +107,11 @@
         "externalDocs": {
           "url": "http://aka.ms/azureautomationsdk/dsccompilationjoboperations"
         },
+        "x-ms-examples": {
+          "Get a DSC Compilation job": {
+            "$ref": "./examples/getCompilationJob.json"
```
Comment (fearthecowboy) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/dscCompilationJob.json]: Validation Failure: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/338705368#L755

**Comment by fearthecowboy** [comment]
Hey @vrdmr -- we're getting closer. Just a few things i need you to look at.

It's a bit tricky, since there are other validation failures and examples failing -- is it possible that whoever owns the examples that are failing could look at those and update this PR?

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json (hunk 1 above)
Comment (vrdmr) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json]: The Spec was incorrect here and I just changed it to reflect what we are sending from the service. And we are in the process of releasing new Automation SDKs based on Swagger and are still in preview mode.

Our last stable release of SDK was based on Hyak.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json (hunk 2 above)
Comment (vrdmr) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json]: This resource is a ProxyResource and does not have all the requirements of the Resource (like Location, Tags etc.). Thus I removed the allOf - Resource and just added the tag.

To make it better, I can go and take this route as mentioned here - https://github.com/Azure/azure-rest-api-specs/pull/2368#discussion_r164938982. By creating a separate Proxy Resource.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json (hunk 3 above)
Comment (vrdmr) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json]: Explained above.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/dscCompilationJob.json (hunk 4 above)
Comment (vrdmr) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/dscCompilationJob.json]: Documenting existing API which wasn't listed earlier.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/dscNodeConfiguration.json (hunk 5 above)
Comment (vrdmr) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/dscNodeConfiguration.json]: It returns only 200 and not 201. As mentioned earlier, we don't have any swagger-based stable clients and are in the process of releasing them.

I'll confirm the response code with Fiddler.

Comment (vrdmr) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/dscNodeConfiguration.json]: It returns only 200 and not 201. As mentioned earlier, we don't have any swagger-based stable clients and are in the process of releasing them.

I'll confirm the response code with Fiddler.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/dscCompilationJob.json (hunk 6 above)
Comment (vrdmr) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/dscCompilationJob.json]: Hey, I checked with the team and I resolved the issue. They had added new flags and it wasn't documented in the spec, which caused the validation failures in DSCCompilationJob. Fixed it.

**Comment by vrdmr** [comment]
@fearthecowboy Hi Garrett, Could you please have a look at this PR. If you have any questions, we can have a quick sync. Thanks.

**Comment by fearthecowboy** [comment]
@vrdmr - I'll get back to it first thing in the morning.

**Comment by fearthecowboy** [comment]
@vrdmr  -- You're still showing validator problems:
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/340682526 -- invalid examples validation
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/340682523 -- a lot of these are flagging because they are marked as azure resources but not showing the required fields.

**Comment by vrdmr** [comment]
Hi @fearthecowboy, For the issues you specified in the build, related to [linting issues](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/340682523), we have active work going on to get them down to 0. There are actively three PRs (#2360, #2466 and this one) from our team and I would be sending one more PR for DSC issues.

For the first set of issues related to x-ms-example, I will push in the changes.

**Review by fearthecowboy — APPROVED** [review]

**Comment by fearthecowboy** [comment]
Merging this -- noting that  other PRs (#2360, #2466) are intended to address some of the linter issues.
